ccl/serverccl: TestTenantWithDecommissionedID failed

ccl/serverccl.TestTenantWithDecommissionedID [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4286790&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4286790&tab=artifacts#/) on master @ [a950be5e3399abfb02a05b665fd81b965cc58d11](https://github.com/cockroachdb/cockroach/commits/a950be5e3399abfb02a05b665fd81b965cc58d11):


```
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).sendLockedWithRefreshAttempts
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:242
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).SendLocked
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:177
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnPipeliner).SendLocked
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go:290
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSeqNumAllocator).SendLocked
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_seq_num_allocator.go:105
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnHeartbeater).SendLocked
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:232
          | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*TxnCoordSender).Send
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_coord_sender.go:521
          | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:913
          | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Send
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:1095
          | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:806
          | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:680
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.catalogQuerier.query
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/catalog_query.go:108
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.lookupDescriptorsUnvalidated
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/catalog_query.go:40
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.GetCrossReferencedDescriptorsForValidation
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/descriptors.go:104
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.(*readValidationDereferencer).DereferenceDescriptors
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/descriptors.go:82
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate.(*collectorState).getMissingDescs
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate/validate.go:351
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate.collectDescriptorsForValidation
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate/validate.go:383
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/validate/validate.go:62
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.lookupDescriptorsAndValidate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/descriptors.go:63
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.MustGetDescriptorsByID
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/descriptors.go:148
          | github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.MustGetDescriptorByID
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv/descriptors.go:160
        Wraps: (16) failed to connect to n1 at 127.0.0.1:46589
        Wraps: (17) initial connection heartbeat failed
          |
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/cockroach/pkg/util/netutil/*netutil.InitialHeartbeatFailedError
        Wraps: (18) rpc error: code = PermissionDenied desc = n4 was permanently removed from the cluster at 2022-02-03 02:41:08.895255857 +0000 UTC; it is not allowed to rejoin the cluster
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *colexecerror.notInternalError (4) *telemetrykeys.withTelemetry (5) *withstack.withStack (6) *errutil.withPrefix (7) *errbase.opaqueWrapper (8) *errutil.withPrefix (9) *errbase.opaqueWrapper (10) *errutil.withPrefix (11) *telemetrykeys.withTelemetry (12) *telemetrykeys.withTelemetry (13) *errbase.opaqueWrapper (14) *errutil.withPrefix (15) *errbase.opaqueWrapper (16) *errutil.withPrefix (17) *errbase.opaqueWrapper (18) *status.Error
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantWithDecommissionedID786684060
--- FAIL: TestTenantWithDecommissionedID (33.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantWithDecommissionedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/serverccl/BUILD.bazel
+++ b/pkg/ccl/serverccl/BUILD.bazel
@@ -9,7 +9,7 @@ go_library(
 
 go_test(
     name = "serverccl_test",
-    size = "large",
+    size = "medium",
     srcs = [
         "admin_test.go",
         "main_test.go",
@@ -25,6 +25,7 @@ go_test(
         "//pkg/ccl/kvccl",
         "//pkg/ccl/utilccl",
         "//pkg/ccl/utilccl/licenseccl",
+        "//pkg/kv/kvserver/liveness",
         "//pkg/kv/kvserver/liveness/livenesspb",
         "//pkg/roachpb:with-mocks",
         "//pkg/security",
@@ -36,7 +37,6 @@ go_test(
         "//pkg/sql/distsql",
         "//pkg/sql/tests",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util",

--- a/pkg/ccl/serverccl/tenant_decommissioned_host_test.go
+++ b/pkg/ccl/serverccl/tenant_decommissioned_host_test.go
@@ -15,10 +15,10 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/livenesspb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
@@ -33,19 +33,21 @@ func TestTenantWithDecommissionedID(t *testing.T) {
 	// the system.sql_instances table. The sql process sets rpcContext.NodeID =
 	// InstanceID and PingRequest.NodeID = rpcContext.NodeID.
 	//
-	// When a KV node recieves a ping, it checks the NodeID against a
+	// When a KV node receives a ping, it checks the NodeID against a
 	// decommissioned node tombstone list. Until PR #75766, this caused the KV
 	// node to reject pings from sql servers. The rejected pings would manifest
 	// as sql connection timeouts.
 
-	skip.UnderStress(t, "decommissioning times out under stress")
-
 	ctx := context.Background()
-	tc := serverutils.StartNewTestCluster(t, 4, base.TestClusterArgs{})
+	tc := serverutils.StartNewTestCluster(t, 1, base.TestClusterArgs{})
 	defer tc.Stopper().Stop(ctx)
 
 	server := tc.Server(0)
-	decommissionID := tc.Server(3).NodeID()
+	hostID := server.NodeID()
+	decommissionID := roachpb.NodeID(int(hostID) + 1)
+
+	liveness := server.NodeLiveness().(*liveness.NodeLiveness)
+	require.NoError(t, liveness.CreateLivenessRecord(ctx, decommissionID))
 	require.NoError(t, server.Decommission(ctx, livenesspb.MembershipStatus_DECOMMISSIONING, []roachpb.NodeID{decommissionID}))
 	require.NoError(t, server.Decommission(ctx, livenesspb.MembershipStatus_DECOMMISSIONED, []roachpb.NodeID{decommissionID}))
 

